code/game/objects/structures/artstuff.dm

```diff
@@ -261,57 +263,68 @@
 	SSpersistence.painting_frames -= src
 
 /obj/structure/sign/painting/attackby(obj/item/I, mob/user, params)
-	if(!C && istype(I, /obj/item/canvas))
+	if(!current_canvas && istype(I, /obj/item/canvas))
 		frame_canvas(user,I)
-	else if(C && C.painting_name == initial(C.painting_name) && istype(I,/obj/item/pen))
+	else if(current_canvas && current_canvas.painting_name == initial(current_canvas.painting_name) && istype(I,/obj/item/pen))
 		try_rename(user)
 	else
 		return ..()
 
 /obj/structure/sign/painting/examine(mob/user)
 	. = ..()
-	if(C)
-		C.ui_interact(user)
+	if(current_canvas)
+		current_canvas.ui_interact(user)
+	. += "<span class='notice'>Any painting placed here will be archived at the end of the shift. Use wirecutters to remove the painting.</span>"
```

Comment:
The part about archiving should be checking for a persistence_id, and still display if there's no canvas currently present